applications/upgradeability-by-proxy.md

```diff
@@ -0,0 +1,187 @@
+# Open Grant Proposal
+
+* **Project Name:** ink! Smart Contract Upgradeability
+* **Team Name:** Fractal
+* **Payment Address:** lolcrypto
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+The main goal of this project is to bring smart contract upgradeability to ink.
+
+Smart contract upgradeability is an important mechanism to fix security bugs in
+contracts. It's also important to have flexibility of adding new features.
+
+This mechanism should be flexible enough that contract developers may decide if
+they want to be able to upgrade their contracts, and specify under what
+conditions.
+
+Our team is interested in this because it is very likely that our development
+efforts start with smart contracts before moving to a custom Substrate runtime.
+Upgradeability is important not just because bugs but because product discovery
+will constantly provide new inputs.
+
+
+### Project Details
+
+Upgrade functionality can be implemented in user-land, or as part of the smart
+contracts pallet. We are starting with the first option. This gives us more
+flexibility to experiment and iterate on our solution. When we're happy with
+the result, we'll see what can be migrated to ink as a directive to reduce
+boilerplate.
```

Comment:
```suggestion
contracts pallet. We are starting with the first option: it gives us more
flexibility to experiment, and iterate faster towards an adequate solucion.
When we're happy with the result, we'll see what can be migrated to ink as a
directive to reduce boilerplate.
```

Reply:
(wasn't sure whether it'd be 100% clear to the reader that "this gives" is connected to "first option")